tests/components/freedompro/test_switch.py

```diff
@@ -0,0 +1,462 @@
+                "type": "carbonDioxideSensor",
+                "state": {"carbonDioxideDetected": False, "carbonDioxideLevel": 0},
+                "online": True,
+            },
+            {
+                "uid": "3WRRJR6RCZQZSND8VP0YTO3YXCSOFPKBMW8T51TU-LQ*3-QURR5Q6ADA8ML1TBRG59RRGM1F9LVUZLKPYKFJQHC",
+                "type": "lightbulb",
+                "state": {"on": False},
+                "online": True,
+            },
+            {
+                "uid": "3WRRJR6RCZQZSND8VP0YTO3YXCSOFPKBMW8T51TU-LQ*TWMYQKL3UVED4HSIIB9GXJWJZBQCXG-9VE-N2IUAIWI",
+                "type": "thermostat",
+                "state": {
+                    "heatingCoolingState": 1,
+                    "currentTemperature": 14,
+                    "targetTemperature": 14,
+                },
+                "online": True,
+            },
+            {
+                "uid": "3WRRJR6RCZQZSND8VP0YTO3YXCSOFPKBMW8T51TU-LQ*3XSSVIJWK-65HILWTC4WINQK46SP4OEZRCNO25VGWAS",
+                "type": "windowCovering",
+                "state": {"position": 0},
+                "online": True,
+            },
+            {
+                "uid": "3WRRJR6RCZQZSND8VP0YTO3YXCSOFPKBMW8T51TU-LQ*JVRAR_6WVL1Y0PJ5GFWGPMFV7FLVD4MZKBWXC_UFWYM",
+                "type": "lightSensor",
+                "state": {"currentAmbientLightLevel": 1},
+                "online": True,
+            },
+            {
+                "uid": "3WRRJR6RCZQZSND8VP0YTO3YXCSOFPKBMW8T51TU-LQ*0PUTVZVJJJL-ZHZZBHTIBS3-J-U7JYNPACFPJW0MD-I",
+                "type": "outlet",
+                "state": {"on": False},
+                "online": True,
+            },
+        ],
+    ):
+        async_fire_time_changed(hass, utcnow() + timedelta(hours=2))
+        await hass.async_block_till_done()
+
+        state = hass.states.get(entity_id)
+        assert state
+        assert state.attributes.get("friendly_name") == "Irrigation switch"
+
+        entry = registry.async_get(entity_id)
+        assert entry
+        assert (
+            entry.unique_id
+            == "3WRRJR6RCZQZSND8VP0YTO3YXCSOFPKBMW8T51TU-LQ*1JKU1MVWHQL-Z9SCUS85VFXMRGNDCDNDDUVVDKBU31W"
+        )
+
+        assert state.state == STATE_ON
+
+
+async def test_switch_set_on(hass, init_integration):
+    """Test set on of the switch."""
+    init_integration
+    registry = er.async_get(hass)
+
+    entity_id = "switch.irrigation_switch"
+    state = hass.states.get(entity_id)
+    assert state
+    assert state.state == STATE_OFF
+    assert state.attributes.get("friendly_name") == "Irrigation switch"
+
+    entry = registry.async_get(entity_id)
+    assert entry
+    assert (
+        entry.unique_id
+        == "3WRRJR6RCZQZSND8VP0YTO3YXCSOFPKBMW8T51TU-LQ*1JKU1MVWHQL-Z9SCUS85VFXMRGNDCDNDDUVVDKBU31W"
+    )
+
+    assert await hass.services.async_call(
+        SWITCH_DOMAIN,
+        SERVICE_TURN_ON,
+        {ATTR_ENTITY_ID: [entity_id]},
+        blocking=True,
+    )
+
+    with patch(
```

Comment:
We don't want to patch the API response here, we want to check that `homeassistant.components.freedompro.put_state` was called with the right parameters.

Maybe try something like:
```py
from unittest.mock import ANY

...

    with patch("homeassistant.components.freedompro.put_state") as mock_put_state:
        assert await hass.services.async_call(
            SWITCH_DOMAIN,
            SERVICE_TURN_ON,
            {ATTR_ENTITY_ID: [entity_id]},
            blocking=True,
        )
    mock_put_state.assert_called_once_with(ANY, ANY, ANY, {"on": True})
```

(I'm not sure if we want to test values of `session`, `api_key`, `unique_id`, so I just left them as ANY.)